modules/account-lib/src/coin/xrp/transaction.ts

```diff
@@ -1,20 +1,159 @@
+import { InvalidTransactionError } from '../baseCoin/errors';
+import { SignedXRPTransaction, TxJSON, Signer } from './iface';
+import { KeyPair } from './keyPair';
+import { initApi } from './utils';
 
 export class Transaction extends BaseTransaction {
+  private _xrpTransaction?: rippleTypes.TransactionJSON;
+  private _signedTransaction?: SignedXRPTransaction;
+  private _sender: string;
+
+  constructor(coinConfig: Readonly<CoinConfig>) {
+    super(coinConfig);
+  }
+
+  /**
+   * Get the list of signatures (if any) produced for this transaction.
+   *
+   * @returns {string[]} list of signatures
+   */
+  get signature(): string[] {
+    if (!this._signedTransaction) {
+      return [];
+    }
+    const decodedTx = binaryCodec.decode(this._signedTransaction?.signedTransaction);
+    if (decodedTx.Signers) {
+      const signers = decodedTx.Signers as unknown as { Signer: Signer }[];
+      return signers.map(({ Signer }) => Signer.TxnSignature);
+    }
+    return [];
+  }
+
+  sender(address: string): void {
+    this._sender = address;
+  }
+
+  /**
+   * Sets xrp transaction.
+   *
+   * @param {rippleTypes.TransactionJSON} tx
+   */
+
+  setXRPTransaction(tx: rippleTypes.TransactionJSON): void {
+    this._xrpTransaction = tx;
+  }
+
+  /**
+   * Get underlaying xrp transaction.
+   *
+   * @returns {rippleTypes.TransactionJSON}
+   */
+
+  getXRPTransaction(): rippleTypes.TransactionJSON | undefined {
+    return this._xrpTransaction;
+  }
+
+  /**
+   * Set the transaction type.
+   *
+   * @param {TransactionType} transactionType The transaction type to be set.
+   */
+  setTransactionType(transactionType: TransactionType): void {
+    this._type = transactionType;
+  }
+
   /** @inheritdoc */
   canSign(key: BaseKey): boolean {
-    throw new NotImplementedError('canSign not implemented');
+    return true;
+  }
+
+  /**
+   * Signs transaction.
+   *
+   * @param {KeyPair} keyPair Signer keys.
+   */
+  sign(keyPair: KeyPair[]): void {
+    if (!this._xrpTransaction) {
+      throw new InvalidTransactionError('Empty transaction');
+    }
+    if (keyPair.length > 0) {
+      const keys = {
+        privateKey: keyPair[0].getKeys().prv,
+        publicKey: keyPair[0].getKeys().pub,
+      };
+      const txJSON = JSON.stringify(this._xrpTransaction);
+      this._signedTransaction = initApi().sign(txJSON, keys, { signAs: keyPair[0].getAddress() });
+    }
   }
 
   /** @inheritdoc */
   toBroadcastFormat(): string {
-    throw new NotImplementedError('toBroadcastFormat not implemented');
+    if (!this._xrpTransaction) {
+      throw new InvalidTransactionError('Empty transaction');
+    }
+    if (this._signedTransaction) {
+      return this._signedTransaction.signedTransaction;
+    } else {
+      return binaryCodec.encode(this._xrpTransaction);
+    }
   }
 
   /** @inheritdoc */
-  toJson(): any {
-    throw new NotImplementedError('toJson not implemented');
+  toJson(): TxJSON {
```

Comment:
Why not `return this._xrpTransaction;`?

Reply:
toJson formats the underlying transaction to a more readable format